go-controller/pkg/util/kube.go

```diff
@@ -265,6 +265,14 @@ func PodScheduled(pod *kapi.Pod) bool {
 	return pod.Spec.NodeName != ""
 }
 
+// PodTerminatingWithFinalizer returns true if the pod is running and is terminating waiting on one or more finalizers
+func PodTerminatingWithFinalizer(pod *kapi.Pod) bool {
+	if pod.Status.Phase == kapi.PodRunning && pod.DeletionTimestamp != nil {
```

Comment:
So I get the deletion timestamp check https://kubernetes.io/docs/concepts/overview/working-with-objects/finalizers/  makes sense...but the state being running.... is that the only case we care about? (thinking out loud in case there is some other pod lifecycle scenario we need to cover here...like succeeded/unknown and finalizers enabled? https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase )

Reply:
oh good point, let me look into this

Reply:
So @jcaamano @tssurya so when deleting the pod it goes to terminating and does not stay in running. However, when i delete the node there is a period of time (like 10-15 seconds in kind), until the pod goes from running to terminating. I think this is why I was confused on this. 

So I'm going to change this patch to just ignore pods in Terminating state (similar behavior as completed).